From 668570754c4155d0f7ac6fff96ffda5836da8c52 Mon Sep 17 00:00:00 2001
From: suman mallela <suman.m@intel.com>
Date: Tue, 18 Dec 2018 11:42:26 +0530
Subject: [PATCH] wpa-supplicant: Handle long P2P Device Interface name

The default P2P Device interface name was constructed

might result with an interface name that exceeds the maximum

allowed interface name leagth (IFNAMSZ).

Fix this by limiting the properly limiting the created
interface name.

Signed-off-by: suman mallela <suman.m@intel.com>
---
 wpa_supplicant/p2p_supplicant.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/wpa_supplicant/p2p_supplicant.c b/wpa_supplicant/p2p_supplicant.c
index b1fdc28..c002ee9 100644
--- a/wpa_supplicant/p2p_supplicant.c
+++ b/wpa_supplicant/p2p_supplicant.c
@@ -3730,14 +3730,20 @@ int wpas_p2p_add_p2pdev_interface(struct wpa_supplicant *wpa_s,
 {
 	struct wpa_interface iface;
 	struct wpa_supplicant *p2pdev_wpa_s;
-	char ifname[100];
-	char force_name[100];
+        char ifname[IFNAMSIZ];
+        char force_name[IFNAMSIZ];
 	int ret;
 
 	ret = os_snprintf(ifname, sizeof(ifname), P2P_MGMT_DEVICE_PREFIX "%s",
 			  wpa_s->ifname);
-	if (os_snprintf_error(sizeof(ifname), ret))
-		return -1;
+        if (os_snprintf_error(sizeof(ifname), ret)) {
+                if (ret < 0)
+                        return -1;
+                wpa_printf(MSG_WARNING,
+                           "P2P: P2P Device interface name truncated=%s",
+                           ifname);
+        }
+
 	force_name[0] = '\0';
 	wpa_s->pending_interface_type = WPA_IF_P2P_DEVICE;
 	ret = wpa_drv_if_add(wpa_s, WPA_IF_P2P_DEVICE, ifname, NULL, NULL,
-- 
2.7.4

